dolphin/alert_manager/alert_processor.py

```diff
@@ -0,0 +1,81 @@
+# Copyright 2020 The SODA Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http:#www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from oslo_log import log
+
+from dolphin import context
+from dolphin import db
+from dolphin import exception
+from dolphin.drivers import api as driver_manager
+
+LOG = log.getLogger(__name__)
+
+
+class AlertProcessor(object):
+    """Alert model translation and export functions"""
+
+    def __init__(self):
+        self.driver_manager = driver_manager.API()
+
+    def process_alert_info(self, alert):
+        """Fills alert model using driver manager interface."""
+
+        ctxt = context.get_admin_context()
+
+        # Trap source ip should be configured as part of alert source
+        # configuration. Incoming source ip will be mapped with Trap source
+        # ip in db and storage id will be obtained As config flow not exists
+        # now Currently source ip is mapped with access_info ip address
+
+        # First retrieve access_info from source ip and get storage info
+        # using storage id
+        # This is a temporary mechanism till the alert config flow is handled
+        filters = {'host': alert['transport_address']}
+        access_info = db.access_info_get_all(ctxt, filters=filters)
```

Comment:
Please open an issue to provide the API for access_info_get based upon the filters. Else this will not be efficient. We will handle it through issues later

Reply:
https://github.com/sodafoundation/SIM/issues/30